src/api.py

```diff
@@ -1,26 +1,100 @@
+from pydantic import BaseModel
+from starlette.responses import HTMLResponse
+
 from src import service
+from src.generators import html_generator
+
+app = FastAPI()
+format_html = False
+
 
-app = Flask(__name__)
+class TaskOut(BaseModel):
+    id: str
+    status: str
+    result: str | None = None
+    status_callback_url: str | None = None
 
 
 # Sample url: https://tmt.org/?test-url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke
 # or for plans: https://tmt.org/?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic
-@app.route("/", methods=["GET"])
-def find_test():
-    test_url = request.args.get("test-url", default=None)
-    test_name = request.args.get("test-name", default=None)
-    test_ref = request.args.get("test-ref", default="main")
+@app.get("/")
+def find_test(
+        test_url: str = Query(None, alias="test-url"),
+        test_name: str = Query(None, alias="test-name"),
+        test_ref: str = Query("default", alias="test-ref"),
+        test_path: str = Query(None, alias="test-path"),
+        plan_url: str = Query(None, alias="plan-url"),
+        plan_name: str = Query(None, alias="plan-name"),
+        plan_ref: str = Query("default", alias="plan-ref"),
+        plan_path: str = Query(None, alias="plan-path"),
+        out_format: str = Query("json", alias="format")
+):
+    # Parameter validations
     if (test_url is None and test_name is not None) or (test_url is not None and test_name is None):
         return "Invalid arguments!"
-    plan_url = request.args.get("plan-url", default=None)
-    plan_name = request.args.get("plan-name", default=None)
-    plan_ref = request.args.get("plan-ref", default="main")
     if (plan_url is None and plan_name is not None) or (plan_url is not None and plan_name is None):
         return "Invalid arguments!"
-    html_page = service.main(test_url, test_name, test_ref, plan_url, plan_name, plan_ref)
-    return html_page
+    if plan_url is None and plan_name is None and test_url is None and test_name is None:
+        return "Missing arguments!"
+        # TODO: forward to docs
+    # Disable Celery if not needed
+    if os.environ.get("USE_CELERY") == "false":
+        return service.main(
+            test_url=test_url,
+            test_name=test_name,
+            test_ref=test_ref,
+            plan_url=plan_url,
+            plan_name=plan_name,
+            plan_ref=plan_ref,
+            out_format=out_format,
+            test_path=test_path,
+            plan_path=plan_path)
+    r = service.main.delay(
+            test_url=test_url,
+            test_name=test_name,
+            test_ref=test_ref,
+            plan_url=plan_url,
+            plan_name=plan_name,
+            plan_ref=plan_ref,
+            out_format=out_format,
+            test_path=test_path,
+            plan_path=plan_path)
+    # Special handling of response if the format is html
+    if out_format == "html":
+        global format_html
```

Comment:
this variable seems to be unused and can be removed (it's getting flipped between true/false, but nothing ever reads it)

Reply:
hmm, yeah, thanks for pointing that out. Removed in https://github.com/teemtee/web/pull/4/commits/eb00be8daf054caa1df0c8de93c019779e9b65b8.  
@tkoscieln fyi

Reply:
Yeah, this seems to be replaced by standalone endpoint for returning status in HTML format